### PR TITLE
tests: common library for 'ok & fatal function'

### PR DESCRIPTION
**Pull request by aaradhak**
A separate common library is created for all the non exclusive external kola tests.
https://issues.redhat.com/browse/GRPA-4478

### CODE REVIEW

**Comment by aaradhak** [comment]
@mike-nguyen Can you review this PR?

File: tests/common/lib.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
```
Comment (mike-nguyen) [inline tests/common/lib.sh]: Maybe add a comment about what this file does and why it exists.

**Review by mike-nguyen — COMMENTED** [review]
I noticed these only cover the tests in the files directory. There are tests in other directories that use the same `ok` and `fatal` functions.

File: tests/common/lib.sh (hunk 1 above)
Comment (aaradhak) [reply tests/common/lib.sh]: ok . I have added the comment

**Comment by dustymabe** [comment]
The need for a symlink (like `../../../common/`) for each test is unfortunate. I wonder if we should expose a new environment variable from cosa like `$KOLA_EXT_CONFIG_TOPLEVEL` so we can drop the need to create the symlinks. @jlebon @mike-nguyen - WDYT?

**Comment by dustymabe** [comment]
if a file exists at `tests/common/lib.sh` won't kola try to execute that as a test itself?

**Comment by jlebon** [comment]
> I wonder if we should expose a new environment variable from cosa like `$KOLA_EXT_CONFIG_TOPLEVEL` so we can drop the need to create the symlinks.

The thing is that not all the repo is synced to the VM, just the test executable and anything in its `data/`. One thing we could do is to have it in `tests/kola/data` and teach kola to copy everything in the `data/` dirs of all its parent dirs instead of just its own dir.

> if a file exists at `tests/common/lib.sh` won't kola try to execute that as a test itself?

`kola` only looks for tests in `tests/kola`. We could also put it in `tests/kola/data` as above and mark it as non-executable (which is more correct anyway).

**Comment by jlebon** [comment]
> One thing we could do is to have it in tests/kola/data and teach kola to copy everything in the data/ dirs of all its parent dirs instead of just its own dir.

I wanted to add, I don't think we should block on this. This is a really nice improvement already!

WDYT about dropping the executable bit and putting it in e.g. `tests/kola/common` for now?

**Comment by dustymabe** [comment]
> WDYT about dropping the executable bit and putting it in e.g. `tests/kola/common` for now?

@jlebon Did you mean `tests/kola/data` here? In https://github.com/coreos/fedora-coreos-config/pull/1364#issuecomment-990086801 the directory you were using the `tests/kola/data` dir in your hypothetical example.

**Comment by jlebon** [comment]
> > WDYT about dropping the executable bit and putting it in e.g. `tests/kola/common` for now?
> 
> @jlebon Did you mean `tests/kola/data` here? In [#1364 (comment)](https://github.com/coreos/fedora-coreos-config/pull/1364#issuecomment-990086801) the directory you were using the `tests/kola/data` dir in your hypothetical example.

Either should be fine for now until we implement the parent `data/` thing. I guess we can put it in `tests/kola/data` as prep for that.

**Comment by dustymabe** [comment]
> Either should be fine for now until we implement the parent `data/` thing. I guess we can put it in `tests/kola/data` as prep for that.

That sounds good to me. Then IIUC all we'd need to do (once that change lands in COSA) is delete the symlinks.

**Comment by dustymabe** [comment]
Thanks for working on this @aaradhak. Here's what I see now:

```
$ tree tests/kola/files/data/
tests/kola/files/data/
└── data -> ../../data/

1 directory, 0 files
```

Instead of having a second `data` dir (the symlink) under `$KOLA_EXT_DATA` would it be better to just symlink the file directly? Also might be good to rename the file something more obvious just if we do this. Here's a concrete suggestion:

```
$ mv tests/kola/data/lib.sh tests/kola/data/commonlib.sh 
$ rm tests/kola/files/data/data
$ ln -s ../../data/commonlib.sh tests/kola/files/data/commonlib.sh
```

Then the symlink looks like:

```
$ tree tests/kola/files/data/
tests/kola/files/data/
└── commonlib.sh -> ../../data/commonlib.sh

0 directories, 1 file
```

I think we probably also want to squash all of these commits down into a single commit before we do a final merge.

**Comment by aaradhak** [comment]
Hi @dustymabe , the lib.sh have been renamed to commonlib.sh and have made the associated changes. Kindly review the changes.

File: tests/kola/data/commonlib.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+
+#!/bin/bash
+#This is a common library created for the ok and fatal function and symlinks added to the data/common in each directory
```
Comment (dustymabe) [inline tests/kola/data/commonlib.sh]:
```suggestion
#!/bin/bash

# This is a common library created for the ok and fatal function and
# symlinks added to the data/common in each directory
```

**Review by dustymabe — COMMENTED** [review]
ok the code changes LGTM. There is one suggestion I made to delete a blank line at the beginning of commonlib.sh and break the comment into two lines.

We also need to squash all of these commits into one. Something like this:

```
tests: common library for ok() & fatal() functions

A separate common library is created for all external kola tests to use.
```

**Review by dustymabe — APPROVED** [review]
LGTM

**Pull request merged**